### PR TITLE
client/core: Use coinIDString everywhere.

### LINKED ISSUE #480: coinID: Reverse printed txid hex.
Txid are reversed when printed for all currently supported coins. The tx hash portion of coinID is not reversed when printed. This can be difficult to reason about when debugging. For example, recently there was an error in the client that printed out as:
```
020-06-11 19:15:23.015 [ERR] CORE: Error getting confirmations for 69256461677233ac018df056207a2e1154a7e6faff0d64f97402f046bcd643c000000000: -5: no information for transaction
```

While dcrwallet shows:
```
2020-06-11 19:15:23.013 [INF] WLLT: Inserting unconfirmed transaction c043d6bc46f00274f9640dfffae6a754112e7a2056f08d01ac33726761642569
```

I propose that the tx hash/txid portion of the coinID be reversed when printing for coins that reverse the txid when printing. That is, just adjusting the stringer `String()` methods.

Maybe even have coinID print out as `txid(reversed):output index(minus leading zeros)` divided by a colon, which would be much easier to read for human eyes. For applicable coins, of course.

### PR DESCRIPTION
**Pull request by JoeGruffins**
Looking through client and server, these were the only two spots I could find where the coinID was not being formatted for logs.

closes #480

### CODE REVIEW

File: client/core/core.go [hunk 1]
```diff
@@ -1166,7 +1165,7 @@ func (c *Core) verifyRegistrationFee(wallet *xcWallet, dc *dexConnection, coinID
 
 	regConfs, err := wallet.Confirmations(coinID)
 	if err != nil {
-		log.Errorf("Error getting confirmations for %s: %v", hex.EncodeToString(coinID), err)
+		log.Errorf("Error getting confirmations for %s: %v", coinIDString(assetID, coinID), err)
```
Comment (chappjc) [inline client/core/core.go:1168]: https://github.com/decred/dcrdex/pull/468 removed this wallet.Confirmations check, and the assetID input arg for `verifyRegistrationFee`.

File: client/core/core.go [hunk 2]
```diff
@@ -1176,7 +1175,7 @@ func (c *Core) verifyRegistrationFee(wallet *xcWallet, dc *dexConnection, coinID
 	trigger := func() (bool, error) {
 		confs, err := wallet.Confirmations(coinID)
 		if err != nil {
-			return false, fmt.Errorf("Error getting confirmations for %s: %v", hex.EncodeToString(coinID), err)
+			return false, fmt.Errorf("Error getting confirmations for %s: %v", coinIDString(assetID, coinID), err)
```
Comment (chappjc) [inline client/core/core.go:1178]: I think that `wallet.AssetID` is the relevant asset ID you can use here. Not sure why it was ever an input to `verifyRegistrationFee` now that I see that.  Perhaps it was just missed.

Comment (JoeGruffins) [reply client/core/core.go]: Replaced all instances of regFeeAssetID with wallet.AssetID.

Comment (chappjc) [reply client/core/core.go]: @itswisdomagain Touches the code you were just in.  Can you have a peek and see if this makes sense.  I'm confused why asset id was ever an input argument.

Comment (itswisdomagain) [reply client/core/core.go]: Yeah, makes good sense. I like the usage of `wallet.AssetID` instead of `regFeeAssetID` or an `assetID` parameter.

**Pull request merged**